### PR TITLE
Issues/135/remove duplicate db keys

### LINKED ISSUE #64: Remove redundant Scoresheet fields for judge information (dependency on #85)
Inputs marked as disabled are not sending data in the ajax data save. Changing to readonly fixes this, will be fixed in upcoming commit.

### LINKED ISSUE #85: Replace Sequelize chained promises with join queries
Due to a version mismatch, it appears join queries to not function properly through sequelize.

Objectives:
1. Upgrade / Downgrade `sequelize`, `pg`, and `sqlite3` packages to versions where all are compatible
2. Replace chained db promises with join queries
3. Attempt to do this purely through versioning and migrations. If this is impossible, a script/plan to migrate production data to the new database schema should also be developed.

### LINKED ISSUE #92: Total flight # needs to reflect total number of entries in flight
The total flight # should reflect the current total of entries in the flight. Currently, it is only the same as the current (last) entry. 

When a new scoresheet is created in a flight, all other scoresheets' "flight total" should be updated in the db to reflect the current number of entries in the flight.

This could also be done by simply counting the length of the flight every time a scoresheet is loaded (both on edit and on generate pdf)

### PR DESCRIPTION
**Pull request by prphntm63**
Ok boiz, buckle up - this one's a doozy
* Fixes 135 (for real this time)
* Fixes #64 
* Fixes #85
* Fixes #92 

Changes:
* Refactored Sequelize models to use associations properly
* Added migration to remove redundant keys and implement Sequelize association keys
  * From `Scoresheet`: All user info (including UserId), all flight info. Changed `flight_key` to `FlightID`, `flight_total` removed (dynamically calculated from Flight)
  * From `Flight`: Changed `created_by` to `UserId`
* Refactored a TON of things to use `UserId` in Flight instead of scoresheet to avoid having a duplicate UserId

Tests:
* Migrations and new schema tested with sqlite3 and Postgres locally

### CODE REVIEW

File: controllers/ScoresheetController.js [hunk 1]
```diff
@@ -26,110 +26,42 @@ function jsonErrorProcessor(err, res) {
     });
   } else {
     debug(err);
+    console.log(err);
```
Comment (PH-Lazurous) [inline controllers/ScoresheetController.js:29]: Need to omit console log.

File: controllers/ScoresheetController.js [hunk 2]
```diff
@@ -179,28 +103,30 @@ scoresheetController.getScoresheetData = function (req, res) {
...
-      delete user.created_at;
-      delete user.updated_at;
-
-      res.json({
-        scoresheet: scoresheetData,
-        flight: flightData,
-        user: user,
-      });
+
+    let user = scoresheetData.Flight.User;
```
Comment (PH-Lazurous) [inline controllers/ScoresheetController.js:114]: Would it make more sense to have the model return a cleaned object here vs us manually deleting everything?

File: controllers/ScoresheetController.js (hunk 1 above)
Comment (prphntm63) [reply controllers/ScoresheetController.js:29]: image

File: controllers/ScoresheetController.js (hunk 2 above)
Comment (prphntm63) [reply controllers/ScoresheetController.js:114]: Yep. Got a handle on how to do that?

**Comment by prphntm63** [comment]
* Added a `sanitize()` instance method on the `User` model
* Removed `console.log()`

**Review by PH-Lazurous — APPROVED** [review]

**Pull request merged**